pkgs/development/tools/analysis/qcachegrind/default.nix

```diff
@@ -0,0 +1,44 @@
+{ stdenv, fetchurl, bash, cmake,
+  qt, graphviz,
+  qmakeHook, makeQtWrapper,
+  perl, python, php
+}:
+
+stdenv.mkDerivation rec {
+  name = "qcachegrind-${version}";
+  version = "16.12.3";
+
+  src = fetchurl {
+    url = "http://download.kde.org/stable/applications/${version}/src/kcachegrind-${version}.tar.xz";
+    sha256 = "109y94nz96izzsjjdpj9c6g344rcr86srp5w0433mssbyvym4x7q";
+  };
+
+  buildInputs = [ qt.qtbase perl python php ];
+
+  nativeBuildInputs = [ qmakeHook makeQtWrapper ];
+
+  postInstall = ''
+     mkdir -p $out/bin
+     cp -p converters/dprof2calltree $out/bin/dprof2calltree
+     cp -p converters/hotshot2calltree.cmake $out/bin/hotshot2calltree
+     cp -p converters/memprof2calltree $out/bin/memprof2calltree
+     cp -p converters/op2calltree $out/bin/op2calltree
+     cp -p converters/pprof2calltree $out/bin/pprof2calltree
+     chmod -R +x $out/bin/
+  '' + stdenv.lib.optionalString (stdenv.isDarwin) ''
+     cp -p cgview/cgview.app/Contents/MacOS/cgview $out/bin/cgview
+     wrapQtProgram $out/bin/cgview
+
+     mkdir -p $out/Applications
+     cp -rp qcachegrind/qcachegrind.app $out/Applications
+     wrapQtProgram $out/Applications/qcachegrind.app/Contents/MacOS/qcachegrind
+     chmod -R +x $out/Applications/qcachegrind.app
```

Comment:
I [linked](https://github.com/orivej/nixpkgs/blob/6c1fc0654ba95c30c8ea2bcc4c6f1da38922d138/pkgs/development/tools/analysis/qcachegrind/default.nix#L29-L35) to what I would expect on Linux: an executable in `$out/bin/qcachegrind` and desktop metadata (for qcachegrind to appear in the launcher) in `$out/share/applications`.